### PR TITLE
fix: Add support for default dimension in `aten.cat`

### LINKED ISSUE #1862: 🐛 [Bug] Encountered bug when using `torch.ops.aten.cat` converter
##  Bug Description

When compiling the [T5-Base Model](https://huggingface.co/t5-base) model via the `aten` path, the following error is encountered:

```
IndexError: tuple index out of range
While executing %cat_default : [#users=1] = call_function[target=torch.ops.aten.cat.default](args = ([%arg0, %arg1, %arg2],), kwargs = {_itensor_to_tensor_meta: {<tensorrt.tensorrt.ITensor object at 0x7ff49c2fae30>: None, <tensorrt.tensorrt.ITensor object at 0x7ff49c2e6e30>: None, <tensorrt.tensorrt.ITensor object at 0x7ff49c2e1f30>: None}})
```

## To Reproduce

1. Initialize model: `T5Model.from_pretrained("t5-base").eval().cuda()`
2. Initialize three input tensors, for example: `torch.randint(0, 1, (1, 14), dtype=torch.int32).to("cuda")` ("input_ids", "attention_mask", "decoder_input_ids")
3. (Optional) Use the `transformers` tools to trace the model via: `transformers.utils.fx.symbolic_trace(model, input_names=["input_ids", "attention_mask", "decoder_input_ids"])`
4. Compile the model using FX

## Expected behavior

Model should compile via the `aten` path

## Environment

- Transformers: 4.26.1
 - Torch-TensorRT Version (e.g. 1.0.0): b3f433a
 - PyTorch Version (e.g. 1.0): 2.1.0.dev20230419+cu117
- CPU Architecture: Intel Xeon CPU
- OS: Ubuntu 20.04
- How you installed PyTorch: pip
- Build command you used: `python setup.py develop`
- Are you using local sources or building from archives: local
- Python version: 3.8.13
- CUDA version: 11.7

### PR DESCRIPTION
**Pull request by gs-olive**
# Description

- Add default `dim=0` to concatenation operator for use cases which do not have a specific concatenation dimension specified
- T5 encounters this error during compilation
- Add test cases to elicit error with default dimension

Fixes #1862
Addresses #1740 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [ x ] My code follows the style guidelines of this project (You can use the linters)
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ x ] I have made corresponding changes to the documentation
- [ x ] I have added tests to verify my fix or my feature
- [ x ] New and existing unit tests pass locally with my changes
- [ x ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/fx/test/converters/aten_op/test_cat_aten.py [hunk 1]
```diff
@@ -53,6 +53,41 @@ def forward(self, x, y):
...
+                return torch.cat((x, y, z))
+
+        inputs = [torch.randn(2, 1, 3), torch.randn(1, 1, 3), torch.randn(3, 1, 3)]
+        self.run_test(
+            Cat(),
+            inputs,
+            expected_ops={torch.ops.aten.cat.default},
+        )
+
+    def test_cat_dynamic_shape_no_dim(self):
```
[the remark this answers is not in the document]
Comment (gs-olive) [reply py/torch_tensorrt/fx/test/converters/aten_op/test_cat_aten.py:68]: Yes, the regular implementation works when the dimension is negative now, and I've re-enabled the tests for that. The new case I added does not need separate tests for negative dim, since it is exclusively for the no-dim-specified case, where the user calls `torch.cat` with the default dimension (0).

**Review by frank-wei — APPROVED** [review]
LGTM!
